### PR TITLE
Add scope support and update cache

### PR DESCRIPTION
**Pull request by omercs**
Scope and additional scope support
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/394%23issuecomment-123166082%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/394%23issuecomment-123187040%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/394%23discussion_r35056889%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/394%23discussion_r35066061%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/394%23discussion_r35071485%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/394%23issuecomment-123166082%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22You%20going%20to%20take%20this%20so%20I%20can%20use%20it%3F%22%2C%20%22created_at%22%3A%20%222015-07-21T05%3A28%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1471883%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/brandwe%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%20%22%2C%20%22created_at%22%3A%20%222015-07-21T06%3A39%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RPangrle%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20678dd3eb018c3a373f24a5f0a83993de60daf7c1%20samples/hello/src/com/microsoft/aad/adal/hello/Constants.java%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/394%23discussion_r35056889%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22should%20demo%20code%20be%20pointing%20to%20PPE%3F%22%2C%20%22created_at%22%3A%20%222015-07-20T23%3A40%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RPangrle%22%7D%7D%2C%20%7B%22body%22%3A%20%22PPE%20is%20the%20only%20setup%20for%20interactive%20flow.%20We%20can%20update%20this%20when%20it%20is%20available%20on%20prod.%22%2C%20%22created_at%22%3A%20%222015-07-21T03%3A02%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/466805%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/omercs%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20samples/hello/src/com/microsoft/aad/adal/hello/Constants.java%3AL32-48%22%7D%2C%20%22Pull%20678dd3eb018c3a373f24a5f0a83993de60daf7c1%20src/src/com/microsoft/aad/adal/AuthenticationContext.java%20739%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/394%23discussion_r35071485%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Is%20multiResource%20still%20a%20thing%3F%22%2C%20%22created_at%22%3A%20%222015-07-21T05%3A49%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RPangrle%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/src/com/microsoft/aad/adal/AuthenticationContext.java%3AL1286-1296%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/RPangrle%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/RPangrle'></a>

- [ ] <a href='#crh-comment-Pull 678dd3eb018c3a373f24a5f0a83993de60daf7c1 samples/hello/src/com/microsoft/aad/adal/hello/Constants.java 5'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/394#discussion_r35056889'>File: samples/hello/src/com/microsoft/aad/adal/hello/Constants.java:L32-48</a></b>
- <a href='https://github.com/RPangrle'></a> should demo code be pointing to PPE?
- <a href='https://github.com/omercs'></a> PPE is the only setup for interactive flow. We can update this when it is available on prod.
- [ ] <a href='#crh-comment-Pull 678dd3eb018c3a373f24a5f0a83993de60daf7c1 src/src/com/microsoft/aad/adal/AuthenticationContext.java 739'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/394#discussion_r35071485'>File: src/src/com/microsoft/aad/adal/AuthenticationContext.java:L1286-1296</a></b>
- <a href='https://github.com/RPangrle'></a> Is multiResource still a thing?
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/394#issuecomment-123166082'>General Comment</a></b>
- <a href='https://github.com/brandwe'></a> You going to take this so I can use it?

<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-android/pull/394?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-android/pull/394?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-android/pull/394?approve=0'></a>&nbsp;<a href='https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/394'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by RandalliLama** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/394'>Refresh</a>.<a href='#crh-update-none'></a>

File: samples/hello/src/com/microsoft/aad/adal/hello/Constants.java [hunk 1]
```diff
@@ -32,16 +32,16 @@
     
     // AAD PARAMETERS
     // https://login.windows.net/tenantInfo
-    static final String AUTHORITY_URL = "https://login.windows.net/common";
+    static final String AUTHORITY_URL = "https://login.windows-ppe.net/common";
```
Comment (RPangrle) [inline samples/hello/src/com/microsoft/aad/adal/hello/Constants.java]: should demo code be pointing to PPE?

Comment (omercs) [inline samples/hello/src/com/microsoft/aad/adal/hello/Constants.java]: PPE is the only setup for interactive flow. We can update this when it is available on prod.

**Comment by brandwe** [comment]
You going to take this so I can use it?

File: src/src/com/microsoft/aad/adal/AuthenticationContext.java [hunk 2]
```diff
@@ -1299,11 +1286,10 @@ private String getTokenHash(String token) {
 
         String mRawIdToken;
 
-
         String mTenantId;
 
-        public RefreshItem(TokenCacheKey keyInCache, AuthenticationRequest request, TokenCacheItem item,
-                boolean multiResource) {
+        public RefreshItem(TokenCacheKey keyInCache, AuthenticationRequest request,
+                TokenCacheItem item, boolean multiResource) {
```
Comment (RPangrle) [inline src/src/com/microsoft/aad/adal/AuthenticationContext.java]: Is multiResource still a thing?

**Comment by RPangrle** [comment]
:shipit:

**Pull request merged**